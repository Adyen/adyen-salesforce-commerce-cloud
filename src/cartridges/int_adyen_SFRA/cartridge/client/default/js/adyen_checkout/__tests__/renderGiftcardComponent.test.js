/**
 * @jest-environment jsdom
 */
const {
  removeGiftCards,
  renderAddedGiftCard,
  renderGiftCardSelectForm,
  showGiftCardWarningMessage,
} = require('../renderGiftcardComponent');
const store = require('../../../../../store');

jest.mock('../../commons');
jest.mock('../../../../../store');

beforeEach(() => {
  document.body.innerHTML = `
        <select id="giftCardSelect"></select>
        <ul id="giftCardUl"></ul>
        <ul id="giftCardsList"></ul>
        <div id="giftCardContainer"></div>
        <div id="giftCardSelectContainer"></div>
        <button id="giftCardAddButton"></button>
        <div id="adyenPartialPaymentsOrder"></div>
      `;
  window.AdyenCheckout = jest.fn(async () => ({
    create: jest.fn(),
    paymentMethodsResponse: {
      storedPaymentMethods: [{ supportedShopperInteractions: ['Ecommerce'] }],
      paymentMethods: [{ type: 'amazonpay' }],
    },
    options: {
      amount: 'mocked_amount',
      countryCode: 'mocked_countrycode',
    },
  }));
  window.Configuration = { amount: 0 };
  store.checkoutConfiguration = {
    session: {
      imagePath: 'test_image_path',
    },
  };
  store.checkout = {
    options: {},
  };
});
describe.only('Render gift card', () => {
  it('should remove entire giftcard container when removeGiftCards is successful', async () => {
    document.body.innerHTML += `
        <div id="biggerContainer">
          <div id="cancelGiftCardContainer"></div>
        </div>
      `;
    store.checkout.paymentMethodsResponse = {
      paymentMethods: {
        filter: jest.fn(() => [
          { type: 'giftcard', brand: 'brand1' },
          { type: 'giftcard', brand: 'brand2' },
        ]),
      },
    };

    const data = {
      resultCode: 'Received',
    };
    $.ajax = jest.fn(({ success }) => {
      success(data);
      return { fail: jest.fn() };
    });

    expect(document.querySelector('#biggerContainer').innerHTML).toContain(
      'cancelGiftCardContainer',
    );
    removeGiftCards();
    expect(document.querySelector('#biggerContainer')).toBeNull;
  });

  it('should show a warning message if more items are added to cart later', async () => {
    document.body.innerHTML = `
      <div class="card-body order-total-summary"></div>
    `;

    showGiftCardWarningMessage();
    const orderTotalSummaryEl = document.querySelector(
      '.card-body.order-total-summary',
    );
    expect(orderTotalSummaryEl).toMatchSnapshot();
    expect(document.querySelector('#giftCardWarningMessage')).toBeTruthy();
  });

  it('should render applied gift cards', async () => {
    const giftCardData = { giftCard: { brand: 'brand', name: 'Name' } };
    store.partialPaymentsOrderObj = giftCardData;
    renderAddedGiftCard(giftCardData);

    const giftCardsList = document.querySelector('#giftCardsList');
    expect(giftCardsList).toMatchSnapshot();
  });

  it('should render a select form', async () => {
    store.checkout.paymentMethodsResponse = {
      paymentMethods: {
        filter: jest.fn(() => [
          { type: 'giftcard', brand: 'mocked_brand1' },
          { type: 'giftcard', brand: 'mocked_brand2' },
        ]),
      },
    };
    renderGiftCardSelectForm();

    const giftCardsSelect = document.querySelector('#giftCardSelect');
    const giftCardUl = document.querySelector('#giftCardUl');
    expect(giftCardsSelect).toMatchSnapshot();
    expect(giftCardUl).toMatchSnapshot();
  });
});
