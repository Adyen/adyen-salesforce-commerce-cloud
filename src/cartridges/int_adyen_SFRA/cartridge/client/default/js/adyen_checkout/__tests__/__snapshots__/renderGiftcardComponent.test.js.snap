// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render gift card should render a select form 1`] = `
<select
  id="giftCardSelect"
>
  <option
    style="visibility: hidden;"
    value="mocked_brand1"
  />
  <option
    style="visibility: hidden;"
    value="mocked_brand2"
  />
</select>
`;

exports[`Render gift card should render a select form 2`] = `
<ul
  id="giftCardUl"
>
  <li
    data-brand="mocked_brand1"
    data-name="undefined"
    data-type="giftcard"
  >
    <span />
    <img
      height="26"
      src="test_image_pathmocked_brand1.png"
      width="40"
    />
  </li>
  <li
    data-brand="mocked_brand2"
    data-name="undefined"
    data-type="giftcard"
  >
    <span />
    <img
      height="26"
      src="test_image_pathmocked_brand2.png"
      width="40"
    />
  </li>
</ul>
`;

exports[`Render gift card should render applied gift cards 1`] = `
<ul
  id="giftCardsList"
>
  <div
    class="gift-card"
  >
    <div
      class="wrapper"
    >
      <div
        class="brand-container"
      >
        <img
          class="gift-card-logo"
          src="test_image_pathbrand.png"
        />
        <p>
          Name
        </p>
      </div>
      <div
        class="gift-card-action"
      />
    </div>
    <div
      class="wrapper"
    >
      <p />
      <strong />
    </div>
  </div>
</ul>
`;

exports[`Render gift card should show a warning message if more items are added to cart later 1`] = `
<div
  class="card-body order-total-summary"
>
  <div
    class="alert alert-warning error-message gift-card-warning-msg"
    id="giftCardWarningMessage"
    role="alert"
  >
    <p
      class="error-message-text"
    />
  </div>
</div>
`;
